tests/PHPStan/Rules/Comparison/ImpossibleCheckTypeGenericOverwriteRuleTest.php

```diff
@@ -0,0 +1,44 @@
+<?php declare(strict_types = 1);
+
+namespace PHPStan\Rules\Comparison;
+
+use PHPStan\Rules\Rule;
+use PHPStan\Testing\RuleTestCase;
+
+/**
+ * @extends RuleTestCase<ImpossibleCheckTypeMethodCallRule>
+ */
+class ImpossibleCheckTypeGenericOverwriteRuleTest extends RuleTestCase
+{
+
+	private bool $treatPhpDocTypesAsCertain;
+
+	public function getRule(): Rule
+	{
+		return new ImpossibleCheckTypeMethodCallRule(
+			new ImpossibleCheckTypeHelper(
+				$this->createReflectionProvider(),
+				$this->getTypeSpecifier(),
+				[],
+				$this->treatPhpDocTypesAsCertain,
+			),
+			true,
+			$this->treatPhpDocTypesAsCertain,
+		);
+	}
+
+	public function testNoReportedErrorOnOverwrite(): void
+	{
+		$this->treatPhpDocTypesAsCertain = false;
```

Comment:
This doesn't need to be a property if you're not testing true/false scenarios. So please put "true" directly in the constructor.

Reply:
fixed